### ISSUE TITLE
Issue with `playwright/valid-title` with string variable test name in loop or read from cases array

### ISSUE THREAD

**Issue by gdarchen** [mskelton/eslint-plugin-playwright#368]
Hello,

I noticed an issue with the `playwright/valid-title` rule when the test name comes from a string variable inside of a loop or read from a cases array.

Here is an example:

```ts
type TestCase = {
  name: string;
  // ...
};

const cases: TestCase[] = [
  {
    name: 'my first test',
    // ...
  },
  {
    name: 'my second test',
    // ...
  },
];

test.describe('Tests with loop', () => {
  for (const { name } of cases) {
    test(name, async () => {
      // ...
      expect(1 + 1).toBe(2);
    });
  }
});
```

It wrongly complains that the title is expected to be a string, which is actually the case.

<img width="431" alt="Image" src="https://github.com/user-attachments/assets/5ce74fab-1640-465a-81c2-52b68713cab3" />

We have the same issue when reading from the `cases` array instead of looping.

<img width="425" alt="Image" src="https://github.com/user-attachments/assets/c5c0d7eb-3c0b-4e39-8b07-37099777cc1c" />
